### PR TITLE
Restore Cancel Button

### LINKED ISSUE #5344: Cancel button broken after #5266
> the latest git code on my server seems to have cancel button not working, and I believe it's likely the choice related work that was done recently that caused it...pre do you mind looking into it?

_[18:49:29]  **Freedom SK**_

I bisected and found 6ac82666966135445b4f7a0732a9f8da270f4ea7 is where the cancel button is no longer presented as an option in battles. @MacChaeger, can you PTAL?

### PR DESCRIPTION
**Pull request by MacChaeger**
Fixes #5344

### CODE REVIEW

File: sim/battle.ts [hunk 1]
```diff
@@ -1022,15 +1022,12 @@ export class Battle extends Dex.ModdedDex {
 		}
 		}
 
-		let allRequestsMade = true;
-		for (const request of requests) {
-			if (!request) allRequestsMade = false;
-		}
+		let allRequestsMade = !requests.some(request => !request);
```
Comment (scheibo) [inline sim/battle.ts]:
```

```suggestion
		let allRequestsMade = requests.every(request => request);
```

File: sim/battle.ts [hunk 2]
```diff
@@ -1022,15 +1022,12 @@ export class Battle extends Dex.ModdedDex {
...
 		for (let i = 0; i < this.sides.length; i++) {
 			const side = this.sides[i];
 			const request = requests[i];
 			if (request) {
-				if (!this.supportCancel || allRequestsMade) request.noCancel = true;
+				if (!this.supportCancel || !allRequestsMade) request.noCancel = true;
```
Comment (scheibo) [inline sim/battle.ts]: Can we add a regression test for this (or fix any cancel tests we already have that should have caught this)?

Comment (MacChaeger) [reply sim/battle.ts]: For the record, this only affected whether or not the cancel button actually appeared client-side. It didn't actually stop users from sending `/choose undo` and cancelling their choice.

Comment (scheibo) [reply sim/battle.ts]: True. So less critical to worry about, but if we could cover it with a test without too much effort itd still be worth doing.

Comment (Zarel) [reply sim/battle.ts]: Also, basically zero users know about `/choose undo`, so this is approximately as critical; it just means the test needs to be designed to account for this case.

Comment (scheibo) [reply sim/battle.ts]: I think the naive way of adding the test would be to add a new method to ``test/assert.js`` which allows for asserting on the latest ``request`` object on the battle log for a side (unless theres an easier way to access the request in a test than searching through the battle log)

Comment (MacChaeger) [reply sim/battle.ts]: Yeah I was just saying that's why the tests passed.

Comment (Zarel) [reply sim/battle.ts]: We should probably rewrite the relevant code to make it easier to extract requests; that sounds less effort than screwing around with `test/assert.js`.

**Pull request merged**